draft-ietf-core-dns-over-coap.md

```diff
@@ -320,42 +320,37 @@ the Content-Format option by the DoC server.
-CoAP responses that carry any valid DNS response use a "2.05 Content"
-response code.
+A DNS response indicates either success or failure in the RCODE of the DNS header (see {{Section 4.1.1 of -dns}}).
+It is RECOMMENDED that CoAP responses that carry a parseable DNS response as defined in {{-dns}} use a "2.05 Content" response code.
 
 CoAP responses using non-successful response codes MUST NOT contain a DNS response
 and MUST only be used on errors in the CoAP layer or when a request does not
 fulfill the requirements of the DoC protocol.
 
-Communication errors with a DNS server (e.g., timeouts) SHOULD be indicated
-by including a SERVFAIL DNS response in a successful CoAP response.
+Communication errors with an upstream DNS server (e.g., timeouts) MUST be indicated by including a DNS response with the appropriate RCODE in a successful CoAP response, i.e., using a 2.xx response code.
 
 A DoC client might try to repeat a non-successful exchange unless otherwise prohibited.
 The DoC client might also decide to repeat a non-successful exchange with a different URI, for instance, when the response indicates an unsupported Content-Format.
 
 ### Support of CoAP Caching {#sec:resp-caching}
 
-For reliability and energy saving measures content decoupling and thus en-route caching on proxies takes a far greater role than it does, e.g., in HTTP.
-Likewise, CoAP utilizes cache validation to refresh stale cache entries without large messages which regularly uses hashing over the message content for ETag generation.
+For reliability and energy saving measures, content decoupling, i.e., en-route caching on proxies, takes a far greater role than it does, e.g., in HTTP.
+Likewise, CoAP utilizes cache validation to refresh stale cache entries to reduce the amount of large response messages.
+For cache validation, CoAP implementations regularly use hashing over the message content for ETag generation.
 As such, the approach to guarantee the same cache key for DNS responses as proposed in DoH ({{-doh}}, section 5.1) is not sufficient and needs to be updated so that the TTLs in the response are more often the same regardless of query time.
 
 The DoC server MUST ensure that any sum of the Max-Age value of a CoAP response and any TTL in the
 DNS response is less or equal to the corresponding TTL received from an upstream DNS server.
 This also includes the default Max-Age value of 60 seconds (see {{-coap}}, section 5.10.5) when no Max-Age option is provided.
 The DoC client MUST then add the Max-Age value of the carrying CoAP response to all TTLs in a DNS response on reception and use these calculated TTLs for the associated records.
 
-The RECOMMENDED algorithm to assure the requirement for the DoC is to set the Max-Age option of a response to the minimum TTL of a DNS response and to subtract this value from all TTLs of that DNS response.
+The RECOMMENDED algorithm to assure the requirement for DoC is as follows:
```

Comment:
as·sure | əˈʃʊr |
verb
1 [reporting verb] tell someone something positively or confidently to dispel any doubts they may have: 

```suggestion
The RECOMMENDED algorithm to meet the requirement for DoC is as follows:
```